readme.md

```diff
@@ -0,0 +1,77 @@
+
+Buildkit is a toolkit for converting source code to build artifacts in an efficient, expressive and repeatable manner.
+
+Key features:
+- Automatic garbage collection
+- Extendable frontend formats
+- Concurrent dependency resolution
+- Efficient instruction caching
+- Build cache import/export
+- Nested build job invocations
+- Distributable workers
+- Multiple output formats
+- Pluggable architecture
+
+
+Read the proposal from https://github.com/moby/moby/issues/32925
+
+#### Quick start
+
+[![asciicinema example](https://asciinema.org/a/gPEIEo1NzmDTUu2bEPsUboqmU.png)](https://asciinema.org/a/gPEIEo1NzmDTUu2bEPsUboqmU)
+
+Buildkit daemon can be built in two different versions: one that uses [containerd](https://github.com/containerd/containerd) for execution and distribution and a standalone version that doesn't have other dependencies apart from [runc](https://github.com/opencontainers/runc). We are open for adding more backends. `buildd` is a CLI utility for running the gRPC API. 
+
+```
+# buildd daemon (choose one)
+go build -o buildd-containerd -tags containerd ./cmd/buildd
+go build -o buildd-standalone -tags standalone ./cmd/buildd
+
+# buildctl utility
+go build -o buildctl ./cmd/buildctl
+```
+
+You can also use `make binaries` that prepares all binaries into the `bin/` directory.
+
+The first thing to test could be to try building buildkit with buildkit. Buildkit provides a low-level solver format that could be used by multiple build definitions. Preparation work for making the Dockerfile parser reusable as a frontend is tracked in https://github.com/moby/moby/pull/33492. As no frontends have been integrated yet we currently have to use a script to generate this low-level definition.
+
+`examples/buildkit/buildkit.go` is a script that defines how to build different configurations of buildkit and its dependencies using the `client` package. Running this script generates a protobuf definition of a build graph. Note that the script itself does not execute any steps of the build.
+
+You can use `buildctl debug dump` to see what data is this definition.
+
+```
+go run examples/buildkit/buildkit.go | buildctl debug dump | jq .
+```
+
+To start building use `buildctl build` command. The script accepts `--target` flag to choose between `containerd` and `standalone` configurations. In standalone mode buildkit binaries are built together with `runc`. In containerd mode, `containerd` binary is built as well from the upstream repo.
+
+```
+go run examples/buildkit/buildkit.go | buildctl build
+```
+
+`buildctl build` will show interactive progress bar by default while the build job is running. It will also show you the path to the trace file that contains all information about the timing of the individual steps and logs.
+
+
+#### Contributing
+
+Running tests:
+
+```
+make test
+```
+
+Updating vendored dependencies:
+
+```
+# update vendor.conf
+make vendor
+```
+
+Validating your updates before submission:
+
+```
+make validate-all
+```
```

Comment:
Will this repository stay in your name, or be moved to the "docker" or "moby" org?

Can you also;

- add a `CONTRIBUTING.md` (like https://github.com/linuxkit/linuxkit/blob/master/CONTRIBUTING.md) and link it from here?
- add a link to the license here (`Released under the [Apache 2.0 license](LICENSE).`)
- fill in the copyright owner, and year in the license
- add a `NOTICE` file (like https://github.com/linuxkit/linuxkit/blob/master/NOTICE)
- add a poule.yml (assuming we'll be using the regular review procedure); https://github.com/linuxkit/linuxkit/blob/master/poule.yml - may have to add the right labels as well
- add a `MAINTAINERS` file (https://github.com/docker/opensource/blob/master/project-template/MAINTAINERS)

We may have to configure leeroy to do the DCO check on PR's :smile:

Let me know if you want me to take care of those